internal/workflow/preprocessing.go

```diff
@@ -129,6 +130,38 @@ func (w *PreprocessingWorkflow) Execute(
 	}
 	result.addEvent(validateStructureEvent)
 
+	// Add PREMIS objects.
+	premisFilePath := filepath.Join(localPath, "metadata", "premis.xml")
+
+	var addPREMISObjects activities.AddPREMISObjectsResult
+	e = temporalsdk_workflow.ExecuteActivity(
+		withLocalActOpts(ctx),
+		activities.AddPREMISObjectsName,
+		&activities.AddPREMISObjectsParams{
+			PREMISFilePath: premisFilePath,
+			ContentPath:    identifySIP.SIP.ContentPath,
+		},
+	).Get(ctx, &addPREMISObjects)
+	if e != nil {
+		return nil, e
+	}
+
+	// Add PREMIS event noting validate structure result.
+	var addPREMISEvent activities.AddPREMISEventResult
```

Comment:
One of the things I learned from @sevein's work on adding telemetry to Enduro is that there is a significant run time cost to scheduling and then running Temporal activities.  Because of this cost I think it would be better to combine all of the "addPREMIS" activities into a single activity, unless there's some reason to keep them separate.

Reply:
Sounds good! I was having them be separate so the PREMIS file state would reflect where the workflow failed, if an error interrupted it, but that's probably not necessary.